--- a/src/nmath.h
+++ b/src/nmath.h
@@ -144,29 +144,7 @@
    We don't report ME_DOMAIN errors as the callers collect ML_NANs into
    a single warning.
  */
-#define ML_WARNING(x, s) { \
-   if(x > ME_DOMAIN) { \
-       char *msg = ""; \
-       switch(x) { \
-       case ME_DOMAIN: \
-	   msg = _("argument out of domain in '%s'\n");	\
-	   break; \
-       case ME_RANGE: \
-	   msg = _("value out of range in '%s'\n");	\
-	   break; \
-       case ME_NOCONV: \
-	   msg = _("convergence failed in '%s'\n");	\
-	   break; \
-       case ME_PRECISION: \
-	   msg = _("full precision may not have been achieved in '%s'\n"); \
-	   break; \
-       case ME_UNDERFLOW: \
-	   msg = _("underflow occurred in '%s'\n");	\
-	   break; \
-       } \
-       MATHLIB_WARNING(msg, s); \
-   } \
-}
+#define ML_WARNING(x, s) accudist_set_flag(x)
 
 /* Wilcoxon Rank Sum Distribution */
 
