--- a/src/nmath.h
+++ b/src/nmath.h
@@ -207,10 +207,13 @@
 attribute_hidden double pnbeta2(double, double, double, double, double, int, int);
 
 int	Rf_i1mach(int);
 
 /* From toms708.c */
 attribute_hidden void bratio(double a, double b, double x, double y,
 	    		     double *w, double *w1, int *ierr, int log_p);
 
+attribute_hidden void wilcox_free(void);
+attribute_hidden void signrank_free(void);
+
 
 #endif /* MATHLIB_PRIVATE_H */
